### ISSUE TITLE
crashes every time I try to launch

### ISSUE THREAD

**Issue by bet** [Tobi823/ffupdater#506]
**Describe the bug**
Every time I try to start it, it crashes

**To Reproduce**
1. start ffupdater, either from launcher, or the f-droid app
**Expected behavior**
see the main page with status of browsers
**Error message from FFUpdater**
Stacktrace:
```
java.lang.IllegalStateException: Can't find download url with regex pattern '\Qhttps://dist.torproject.org/torbrowser/\E([0-9.]{4,})\Q/tor-browser-\E[0-9.]{4,}+\Q-android-aarch64-multi.apk\E'.
	at de.marmaro.krt.ffupdater.app.impl.TorBrowser.findVersionAndDownloadUrl(TorBrowser.kt:72)
	at de.marmaro.krt.ffupdater.app.impl.TorBrowser.access$findVersionAndDownloadUrl(TorBrowser.kt:26)
	at de.marmaro.krt.ffupdater.app.impl.TorBrowser$findVersionAndDownloadUrl$1.invokeSuspend(TorBrowser.kt:0)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7842)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@d183448, Dispatchers.Main]

```
Logs:
```
java.lang.IllegalStateException: Can't find download url with regex pattern '\Qhttps://dist.torproject.org/torbrowser/\E([0-9.]{4,})\Q/tor-browser-\E[0-9.]{4,}+\Q-android-aarch64-multi.apk\E'.
	at de.marmaro.krt.ffupdater.app.impl.TorBrowser.findVersionAndDownloadUrl(TorBrowser.kt:72)
	at de.marmaro.krt.ffupdater.app.impl.TorBrowser.access$findVersionAndDownloadUrl(TorBrowser.kt:26)
	at de.marmaro.krt.ffupdater.app.impl.TorBrowser$findVersionAndDownloadUrl$1.invokeSuspend(TorBrowser.kt:0)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7842)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@d183448, Dispatchers.Main]

```
Device information:
| Key | Value |
| --- | --- |
| FFUpdater version | 78.2.4 (161) release Github |
| Device | Pixel 3 (blueline, blueline, blueline) |
| Manufacturer | google (Google) | 
| Supported ABIs | arm64-v8a, armeabi-v7a, armeabi |
| Android version | 12 (SDK: 31) |
| OS | abfarm-release-rbe-64-00083, android-build, release-keys, 1640393541000 |

**Comment by bet** [comment]
P.S. force stop and clear app storage didn't fix it

**Comment by bet** [comment]
force stop and clear app storage didn't help

**Comment by ZTenski** [comment]
Same issue here. appears to crash every time it reaches the TOR browser and attempts to fetch the moat revent version, same stack trace as above. I assumre the TOR site is partially down or something given the error, but this exception should probably be caught as it cases the entire app to crash.

**Comment by eleius** [comment]
Same issue here too, using firefox and tor, device not rooted.

**Comment by AlexOnFOSS** [comment]
Same here too. Android 12

Crash report:


`Stacktrace:
```
java.lang.IllegalStateException: Can't find download url with regex pattern '\Qhttps://dist.torproject.org/torbrowser/\E([0-9.]{4,})\Q/tor-browser-\E[0-9.]{4,}+\Q-android-aarch64-multi.apk\E'.
	at de.marmaro.krt.ffupdater.app.impl.TorBrowser.findVersionAndDownloadUrl(TorBrowser.kt:72)
	at de.marmaro.krt.ffupdater.app.impl.TorBrowser.access$findVersionAndDownloadUrl(TorBrowser.kt:26)
	at de.marmaro.krt.ffupdater.app.impl.TorBrowser$findVersionAndDownloadUrl$1.invokeSuspend(TorBrowser.kt:0)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at android.os.Handler.handleCallback(Handler.java:978)
	at android.os.Handler.dispatchMessage(Handler.java:104)
	at android.os.Looper.loopOnce(Looper.java:238)
	at android.os.Looper.loop(Looper.java:357)
	at android.app.ActivityThread.main(ActivityThread.java:8111)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1026)
	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@931b19f, Dispatchers.Main]

```
Logs:
```
java.lang.IllegalStateException: Can't find download url with regex pattern '\Qhttps://dist.torproject.org/torbrowser/\E([0-9.]{4,})\Q/tor-browser-\E[0-9.]{4,}+\Q-android-aarch64-multi.apk\E'.
	at de.marmaro.krt.ffupdater.app.impl.TorBrowser.findVersionAndDownloadUrl(TorBrowser.kt:72)
	at de.marmaro.krt.ffupdater.app.impl.TorBrowser.access$findVersionAndDownloadUrl(TorBrowser.kt:26)
	at de.marmaro.krt.ffupdater.app.impl.TorBrowser$findVersionAndDownloadUrl$1.invokeSuspend(TorBrowser.kt:0)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at android.os.Handler.handleCallback(Handler.java:978)
	at android.os.Handler.dispatchMessage(Handler.java:104)
	at android.os.Looper.loopOnce(Looper.java:238)
	at android.os.Looper.loop(Looper.java:357)
	at android.app.ActivityThread.main(ActivityThread.java:8111)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1026)
	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@931b19f, Dispatchers.Main]

```
Device information:
| Key | Value |
| --- | --- |
| FFUpdater version | 78.2.4 (161) release F-Droid/other |
| Device | moto g32 (devon_gen, devon, devon) |
| Manufacturer | motorola (motorola) | 
| Supported ABIs | arm64-v8a, armeabi-v7a, armeabi |
| Android version | 12 (SDK: 31) |
| OS | ilclbld187, hudsoncm, release-keys, 1693995898000 |         

`

**Comment by eleius** [comment]
> appears to crash every time it reaches the TOR browser and attempts to fetch the moat revent version

in fact:
"we've made the naming scheme for all our build outputs mutually consistent... the names of all our build artifacts should follow the format ${ARTIFACT}-${OS}-${ARCH}-${VERSION}.${EXT}."
https://blog.torproject.org/new-release-tor-browser-130/

> this exception should probably be caught as it causes the entire app to crash

+1

**Comment by Wojtaz0w** [comment]
Thats because TB got updated to version 13 (115esr), it must cause some disturbances!

**Comment by Usland123** [comment]
same issue when it checks for Tor Browser updates

**Comment by Tobi823** [comment]
@eleius Thanks for the link. I will add the Tor Blog to my RSS Reader.

**Comment by Tobi823** [comment]
I found the issue and fixed it in version 78.2.5. To update FFUpdater you can:
- download the APK file from Github/Gitlab () and install it
   - https://github.com/Tobi823/ffupdater/releases/tag/78.2.5
   - https://gitlab.com/Tobiwan/ffupdater_gitlab/-/releases/78.2.5
- use F-Droid with my own repository to update FFUpdater
- temporary uninstall Tor Browser + update FFUpdater in FFUpdater

**Comment by bet** [comment]
Thank you for the fast fix, and detailed explanation of how to get it without waiting for the f-droid rebuild!
